src/cmd/vm.rs

```diff
@@ -20,136 +27,231 @@ pub struct Args {
-    android_build: Option<String>,
+    /// path to betty VM image to start. It has priority over --board and
+    /// --version (they will be ignored)
+    #[argh(option)]
+    vm_image: Option<String>,
 
-    /// extra arguments
+    /// extra arguments to pass to betty.sh. You can pass other options like
+    /// --extra-args "options".
     #[argh(option)]
-    extra_args: Vec<String>,
+    extra_args: Option<String>,
 }
 
 fn run_start(args: &ArgsStart) -> Result<()> {
-    let mut vec = Vec::new();
-    vec.append(&mut vec!["--board", &args.board]);
+    let dir = find_betty_script(&args.arc)?;
+
+    let mut options = Vec::new();
+    options.append(&mut vec!["--board", &args.board]);
     if !args.reuse_disk_image {
-        vec.append(&mut vec!["--reset_image"])
+        options.append(&mut vec!["--reset_image"]);
     }
-    if let Some(display) = &args.display {
-        vec.append(&mut vec!["--display", display]);
+    if args.rootfs_verify {
+        options.append(&mut vec!["--nodisable_rootfs"]);
     }
     if let Some(version) = &args.version {
-        vec.append(&mut vec!["--release", version]);
+        options.append(&mut vec!["--release", version]);
+    }
+    if let Some(vm_image) = &args.vm_image {
+        options.append(&mut vec!["--vm_image", vm_image]);
     }
-    if let Some(android_build) = &args.android_build {
-        vec.append(&mut vec!["--android_build", android_build]);
+    options.append(&mut vec!["--display", "none"]);
+    if let Some(extra_args) = &args.extra_args {
+        options.append(&mut vec![extra_args]);
     }
 
-    let mut options: Vec<String> = vec.iter().map(|s| s.to_string()).collect();
+    let arg = options.join(" ");
+    run_betty(&dir, "start", &arg)?;
 
-    let extra_args = &args.extra_args;
-    options.append(&mut extra_args.clone());
+    println!("To connect the betty instance, run `lium dut shell --dut localhost:9222`.");
+    println!("To push an Android build a betty VM, run `lium arc flash`.");
 
     Ok(())
 }
 
-#[derive(FromArgs, PartialEq, Debug)]
-/// pushes an Android build a running betty instance
-#[argh(subcommand, name = "push")]
-pub struct ArgsPush {
-    /// the android version to push. This is passed to push_to_device.py. e.g.
-    /// cheets_x86/userdebug/123456
-    #[argh(option, short = 'a')]
-    android_build: String,
-
-    /// extra arguments
-    #[argh(option)]
-    extra_args: Vec<String>,
+fn find_betty_script(arc: &Option<String>) -> Result<String> {
+    let path = match arc {
+        Some(p) => p.to_string(),
+        None => env::current_dir()?.to_string_lossy().to_string(),
+    };
+
+    if Path::new(&format!("{}/betty.sh", path)).exists() {
+        return Ok(path);
+    }
+
+    bail!("betty.sh doesn't exist in {path}. Please consider specifying --repo option.")
 }
 
-fn run_push(args: &ArgsPush) -> Result<()> {
-    let vec = ["android_build", &args.android_build];
+fn run_betty(dir: &str, cmd: &str, opts: &str) -> Result<()> {
```

Comment:
replace cmd: &str with an enum, we shouldn't use "stringly typed" arguments when we can avoid it.

You can implement (or derive) Display for enums automatically using [strum](https://crates.io/crates/strum)

Reply:
opts can also be an enum, but it might be a little harder